### PR TITLE
improve event revision accuracy in redis, to avoid revisions duplications for an aggregate

### LINKED ISSUE #90: [redis] aggregates versioning might be inconsistent
While I was trying to use snapshots with redis, I ran into an issue which causes bugs when reloading my stream.
In some cases, and often under heavy load, redis store can produce multiple events with the same revision. So at a moment, the number of events (eg. 15) is higher than the current revision (eg 13, event revision 5 and 6 are "duplicated").
The problem starts when loading a snapshot and the corresponding events. The last snapshot is created with revision 13. And when the snapshot is loaded, events with revision above 13 are loaded with it, so mainly the 2 latest events (revisions 12 and 13), causing an inconsistent state when reloading from snapshot and events.

The problem is caused by the fact that the `streamRevision` of an event is based on the last loaded event, here https://github.com/adrai/node-eventstore/blob/663fb9fc3d0765ae22196c1bff3ea9aa8e79b305/lib/eventstore.js#L402-L413
Because this operation is not atomic, under heavy load or with lots of events, another command on the aggregate can cause the load of the aggregate with the same events, before the first one had enough time to commit. Both generated events finally get the same revision.

The problem is not directly in redis store. And I don't know if other stores face the same issue (using transactions ?). There is no transaction in redis, only atomic operations using `MULTI` (with no dependency on intermediate results) or Lua scripts (when needing to execute an operation depending on the previous result).

I open this ticket to open the discussion on this. I was thinking about a solution, but only for redis: instead of calling directly a `SET`, we could create a Lua script (easily deployed with redis driver, with all the caching stuff) which could `INCR` a value corresponding to the actual revision of an aggregate, and use it as the `streamRevision`, and then make the `SET`. Can't be safer, all is done atomically. And hopefully, redis loads natively a JSON library for Lua. But the problem is that the stream loaded in memory is no longer in sync. It could causes  

Don't hesitate to contribute to this thread. I would love to hear your thoughts. In the mean time, I'll try to make a PR to show my intent.

### PR DESCRIPTION
**Pull request by rehia**
instead of using last event revision, redis use a lua script so that revision is always accurate
Fixes #90

### CODE REVIEW

**Comment by rehia** [comment]
All tests were passing with redis.
Don't hesitate to make your remarks.

**Comment by adrai** [comment]
They don't seem to pass :-( https://travis-ci.org/adrai/node-eventstore/jobs/173308496
Can you enable redis here too? https://github.com/adrai/node-eventstore/blob/master/test/storeTest.js#L7

**Comment by rehia** [comment]
oh yes sure, I'll fix it, sorry. 
I thought redis tests were muted because of the lack of support in Travis.

File: test/eventstoreTest.js [hunk 1]
```diff
@@ -821,7 +821,11 @@ describe('eventstore', function () {
                   snapshotsTableName: 'snapshots' + token
               }
             }
-
+            if (type === 'redis') {
+              options = {
+                db: 3
```
Comment (adrai) [inline test/eventstoreTest.js]: it's ok, just to know it ;-)

**Pull request merged**